core/src/test/scala/dagr/core/tasksystem/TaskTest.scala

```diff
@@ -78,62 +78,58 @@ class TaskTest extends UnitSpec with LazyLogging {
    * Example #1: a "lazy task" that doesn't have all of its required inputs until
    * a previous task on which it is dependent has getTasks called.
    */
+  object Callbacks {}
   {
 
-    "Dagr" should "support a task data connector between two tasks set in getTasks" in {
+    "Dagr" should "support a linker between two tasks set in getTasks" in {
       val firstTask = new FirstTaskNeedsBuilding()
       val lazyTask = new LazyTask(sleepTime = None)
-      Callbacks.connect(lazyTask, firstTask)((l,f) => l.sleepTime = f.sleepTime)
+      firstTask ==> Linker(firstTask, lazyTask){(f,l) => l.sleepTime = f.sleepTime} ==> lazyTask
+
       // so now lazyTask depends on the number from firstTask being not None
-      firstTask.invokeCallbacks()
       firstTask.applyResources(ResourceSet.infinite)
       firstTask.sleepTime should be ('defined)
-      lazyTask.invokeCallbacks()
+      firstTask.tasksDependingOnThisTask.foreach(t => t.asInstanceOf[InJvmTask].inJvmMethod())
       lazyTask.getTasks should have size 1
     }
 
     it should "fail if a task data connector set in getTasks is not yet available" in {
       val firstTask = new FirstTaskNeedsBuilding()
       val lazyTask = new LazyTask(sleepTime = firstTask.sleepTime)
-      Callbacks.connect(lazyTask, firstTask)((l,f) => l.sleepTime = f.sleepTime)
-      // So now lazyTask depends on the number from firstTask being not None.
-      // If we call lazyTask.getTasks, it will call firstTask.getNumber, which will fail,
-      // since fistTask.number is None.
-      lazyTask.invokeCallbacks()
-      an[RuntimeException] should be thrownBy lazyTask.applyResources(ResourceSet.infinite)
+
+      // Linker is purposefully backwards
```

Comment:
add to the comment that you *set it up to fail* not that the Linker is inherently supposed to be used backwards.  I read tests to figure how to use things sometimes, and this would confuse me.